### ISSUE TITLE
KinesisAnalyticsFirehoseInputPreprocessingEvent missing JsonPropertyName attribute on the Base64EncodedData

### ISSUE THREAD

**Issue by TimVanEgmond** [aws/aws-lambda-dotnet#862]
## Description

The KinesisAnalyticsFirehoseInputPreprocessingEvent is missing the JsonPropertyName attribute on the Base64EncodedData property making it incompatible with System.Text.Json

The RecordMetadata is also missing the JsonPropertyName, and that one isn't present in the KinesisAnalyticsStreamsInputPreprocessingEvent either.

## Reproduction Steps

1. Create lambda function that receives a `KinesisAnalyticsFirehoseInputPreprocessingEvent` using the `Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer`
2. Invoke the FirehoseRecord.DecodeData() method
3. An ArgumentNullException is thrown

### Logs

```
Value cannot be null. (Parameter 's'): ArgumentNullException
   at System.Convert.FromBase64String(String s)
   at Amazon.Lambda.KinesisAnalyticsEvents.KinesisAnalyticsFirehoseInputPreprocessingEvent.FirehoseRecord.DecodeData()
```

### Environment

* Build Version:  `Amazon.Lambda.KinesisAnalyticsEvents v2.2.0` `Amazon.Lambda.Serialization.SystemTextJson v2.1.0`
* OS Info: 
* Build Environment:  dotnet cli using the `Amazon.Lambda.Tools v5.1.2`
* Targeted .NET Platform:  .NET Core 3.1

## Resolution
- [ ] :wave: I can/would-like-to implement a fix for this problem myself

---

This is a :bug: bug-report

**Comment by ashishdhingra** [comment]
Hi @TimVanEgmond,

Good afternoon.

I'm not sure about using `Amazon.Lambda.Serialization.SystemTextJson.LambdaJsonSerializer`, but I'm unable to reproduce the issue using the default Visual Studio AWS template that uses `Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer`. I used Mock Lambda Test Tool to send event JSON for Kinesis Analytics.
**.csproj**
```XML
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
    <AWSProjectType>Lambda</AWSProjectType>

    <!-- This property makes the build directory similar to a publish directory and helps the AWS .NET Lambda Mock Test Tool find project dependencies. -->
    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Amazon.Lambda.Core" Version="2.0.0" />
    <PackageReference Include="Amazon.Lambda.KinesisAnalyticsEvents" Version="2.2.0" />
    <PackageReference Include="Amazon.Lambda.Serialization.SystemTextJson" Version="2.1.0" />
  </ItemGroup>
</Project>
```
**Function.cs**
```C#
using Amazon.Lambda.Core;
using Amazon.Lambda.KinesisAnalyticsEvents;
using System.Collections.Generic;

// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]

namespace TestLambdaKinesisFirehouse_Issue862
{
    public class Function
    {
        public KinesisAnalyticsInputPreprocessingResponse FunctionHandler(KinesisAnalyticsStreamsInputPreprocessingEvent evnt, ILambdaContext context)
        {
            context.Logger.LogLine($"InvocationId: {evnt.InvocationId}");
            context.Logger.LogLine($"StreamArn: {evnt.StreamArn}");
            context.Logger.LogLine($"ApplicationArn: {evnt.ApplicationArn}");

            var response = new KinesisAnalyticsInputPreprocessingResponse
            {
                Records = new List<KinesisAnalyticsInputPreprocessingResponse.Record>()
            };

            foreach (var record in evnt.Records)
            {
                context.Logger.LogLine($"\tRecordId: {record.RecordId}");
                context.Logger.LogLine($"\tShardId: {record.RecordMetadata?.ShardId}");
                context.Logger.LogLine($"\tPartitionKey: {record.RecordMetadata?.PartitionKey}");
                context.Logger.LogLine($"\tRecord ApproximateArrivalTime: {record.RecordMetadata?.ApproximateArrivalTimestamp}");
                context.Logger.LogLine($"\tData: {record.DecodeData()}");

                // Add your record preprocessig logic here.

                var preprocessedRecord = new KinesisAnalyticsInputPreprocessingResponse.Record
                {
                    RecordId = record.RecordId,
                    Result = KinesisAnalyticsInputPreprocessingResponse.OK
                };
                preprocessedRecord.EncodeData(record.DecodeData().ToUpperInvariant());
                response.Records.Add(preprocessedRecord);
            }
            return response;
        }
    }
}
```

Also none of the event classes in our repository have attributes marked with `JsonPropertyName`. So I'm not sure if this is a `bug` or a `feature-request`.

Thanks,
Ashish

**Comment by TimVanEgmond** [comment]
Hi @ashishdhingra 

Thanks for taking the time to look into my issue. 

First of all I noticed I made a mistake in my original description. I am using the `Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer` and not the `Amazon.Lambda.Serialization.SystemTextJson.LambdaJsonSerializer`. 

I tried the Function you sent and that one is working for me. But I noticed one difference compared to what I am using, its using the `KinesisAnalyticsStreamsInputPreprocessingEvent` and not the `KinesisAnalyticsFirehoseInputPreprocessingEvent`. Looking at the source of these two event types they are pretty much the same except for the metadata. Changing the input type to what I was using before does throw the exception again. Like I said in my original post I expect this to be because the Firehose event does not have the JsonPorpertyName that the Streams event does [here](https://github.com/aws/aws-lambda-dotnet/blob/20046865db2afd3cf57b1d92d492f1fb4ec08a9d/Libraries/src/Amazon.Lambda.KinesisAnalyticsEvents/KinesisAnalyticsStreamsInputPreprocessingEvent.cs#L135) but only for the netcoreapp3.1 build.

I also tried the function using the Newtonsoft json serializer and that is working fine because newtonsoft also takes the DataContract attribute into account. Because of this I would assume the RecordMetadata to always be null as well when using SystemTextJson.
